articles/contributing/style-guide.md

```diff
@@ -0,0 +1,157 @@
+In such cases, the action taken by the input operation is not clear when the outer operation is defined, such that the right verb is not immediately clear.
+We recommend the verb `Apply`, as in `ApplyIf`, `ApplyToEach`, and `ApplyToFirst`.
+Other verbs may be useful in this case as well, as in `IterateThroughCartesianPower`.
+
+| Verb | Expected Effect |
+| ---- | ------ |
+| Apply | An operation provided as input is called |
+| Estimate | A classical value is returned, representing an estimate drawn from one or more measurements |
+| Measure | A quantum measurement is performed, and its result is returned to the user |
+| Prepare | A given register of qubits is initialized into a particular state |
+| Sample | A classical value is returned at random from some distribution |
+
+For functions, we suggest avoiding the use of verbs in favor of common nouns (see guidance on proper nouns below) or adjectives:
+
+- `ConstantArray`
+- `Head`
+- `LookupFunction`
+
+In particular, we suggest using past participles where appropriate to indicate that a function name is strongly connected to an action or side effect elsewhere in a quantum program.
+For example,  `ControlledOnInt` uses the part participle form of the verb "control" to indicate that the function acts as an adjective to modify its argument.
+This name has the additional benefit of matching the semantics of the built-in `Controlled` functor, as discussed further below.
+
+Similarly, _agent nouns_ can be used to construct function and UDT names from operation names, as in the case of the name `Encoder` for a UDT that is strongly associated with `Encode`. 
+
+### Shorthand ###
+
+The above advice notwithstanding, there are many forms of shorthand that see common and pervasive use in quantum computing.
+We suggest using existing and common shorthand where it exists, especially for operations that are intrinsic to the operation of a target machine.
+For example, we choose the name `X` instead of `ApplyX`, and `Rz` instead of `RotateAboutZ`.
+
+### Proper Nouns in Names ###
+
+
+
+- Avoid using people's names in operation and function names where reasonable.
+  Consider using names that describe the implemented functionality;
+  e.g. `CCNOT` versus `Toffoli` or `CSWAP` versus `Fredkin `.
+  In sample code, consider using names that are familiar to the community reading each particular example, even if that would otherwise run counter to these suggestions.
+  **NB:** names should still appear in documentation comments.
+
+
+
+- If an operation or function is not intended for direct use, but rather should be used by a matching callable which acts by partial application, consider using a name ending with `Impl` for the callable that is partially applied.
+  By contrast, if an operation or function should never be directly called by a user, consider indicating this with a leading `_`.
+- Value argument names should be descriptive; avoid one or two letter names where possible.
```

Comment:
Without autocomplete it is a pain to retype "queryRegister" and "targetRegister" every time instead of "x" and "y". Not that I'm advocating for unreadable variable names...

Reply:
This just pulls over the existing style guide with only formatting changes to make DocFX happy. If it's alright, I'd like to defer more in-depth discussion for a PR on the new style guide, as that has a lot more details than the old one.